lib/spack/spack/package.py

```diff
@@ -1189,7 +1189,7 @@ def extendee_spec(self):
         deps = []
 
         # If the extendee is in the spec's deps already, return that.
-        for dep in self.spec.traverse(deptypes=('link', 'run')):
+        for dep in self.spec.traverse(deptype=('link', 'run')):
```

Comment:
This is also hard to test. I don't think:
```python
extends('python', type='build')
```
would ever make sense in the first place.

Reply:
Yeah, but you can still write it in a test package, there's nothing to prevent it currently.

Reply:
Sure, but what would be the test actually "testing"? I'd like to tie it to a use case or a legit use of the API, rather than to an implementation detail.